Q1.java

```diff
@@ -0,0 +1,132 @@
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+/** Assignment 2 Question 1: Print all ancestors of a node in a binary tree
+ */
+public class Q1 {
+	
+	/** allAncestors() returns a string with comma separated list of 
+	 * ancestors of a given number "num" in a binary tree "t"
+	 * 	
+	 * @param t
+	 * @param num
+	 * @return
+	 */
+	public static String allAncestors(Tree t, int num){
+		StringBuilder str = new StringBuilder();
+		if(t == null){
+			str.append("Tree not initiated");
+			return str.toString();
+		}
+		if(t.root == null){
+			str.append("Empty tree");
+			return str.toString();
+		}
+		if(!allAncestorHelper(t.root, num, str)){
+			str.append(num+" not found");
+		}
+		return str.toString();
+	}
+	
+	static boolean allAncestorHelper(Node<Integer> root, int num, StringBuilder str){

```

Comment:
allAncestorHelper is a complex method. Adding a comment explaining what will be appended to str would be helpful